src/main/java/org/wildfly/security/http/impl/FormAuthenticationMechanism.java

```diff
@@ -205,6 +206,16 @@ private void attemptAuthentication(HttpServerRequest request) throws HttpAuthent
     }
 
     private boolean authorize(String username, HttpServerRequest request) throws HttpAuthenticationException {
+        if (log.isTraceEnabled()) {
+            try {
+                log.tracef("Trying to authorize user %s using FormAuthenticationMechanism. Request URI: [%s], Context path: [%s], Session ID: [%s]",
+                        username, request.getRequestURI(), this.contextPath, getSessionScope(request, false).getID());
+            } catch (NullPointerException ex) {
```

Comment:
Using an NPE catch as control flow is not a good pattern.  It's better to do the null check by hand.